routes/tasks.js

```diff
@@ -6,16 +6,38 @@ const { createTask, updateTask, updateSelfTask, getTasksValidator } = require(".
 const authorizeRoles = require("../middlewares/authorizeRoles");
 const { APPOWNER, SUPERUSER } = require("../constants/roles");
 const assignTask = require("../middlewares/assignTask");
-const cache = require("../utils/cache");
+const { cache, invalidateCache } = require("../utils/cache");
+const { TASKS_ALL } = require("../constants/cacheKeys");
 
-router.get("/", getTasksValidator, cache(), tasks.fetchTasks);
+router.get("/", getTasksValidator, cache({ invalidationKey: TASKS_ALL, expiry: 10 }), tasks.fetchTasks);
 router.get("/self", authenticate, tasks.getSelfTasks);
 router.get("/overdue", authenticate, authorizeRoles([SUPERUSER]), tasks.overdueTasks);
-router.post("/", authenticate, authorizeRoles([APPOWNER, SUPERUSER]), createTask, tasks.addNewTask);
-router.patch("/:id", authenticate, authorizeRoles([APPOWNER, SUPERUSER]), updateTask, tasks.updateTask);
+router.post(
+  "/",
+  authenticate,
+  authorizeRoles([APPOWNER, SUPERUSER]),
+  invalidateCache({ invalidationKeys: [TASKS_ALL] }),
+  createTask,
+  tasks.addNewTask
+);
+router.patch(
+  "/:id",
+  authenticate,
+  authorizeRoles([APPOWNER, SUPERUSER]),
+  invalidateCache({ invalidationKeys: [TASKS_ALL] }),
```

Comment:
this object `{ invalidationKeys: [TASKS_ALL] }` is used at multiple places . can we make it a constant and use it in all routes

Reply:
I agree it's too verbose, but if we have a situation in the future where only one API needs to invalidate some other key. It will lead to inconsistent code style.